anyway/infographics_utils.py

```diff
@@ -978,29 +1019,352 @@ def top_road_segments_accidents_mock_data():  # Temporary for Frontend
+    if widget_id == WidgetId.accident_count_by_severity:
+        res = AccidentCountBySeverityWidget(request_params)
+    elif widget_id == WidgetId.most_severe_accidents_table:
+        res = MostSevereAccidentsTableWidget(request_params)
+    elif widget_id == WidgetId.most_severe_accidents:
+        res = MostSevereAccidentsWidget(request_params)
+    elif widget_id == WidgetId.street_view:
+        res = StreetViewWidget(request_params)
+    elif widget_id == WidgetId.head_on_collisions_comparison:
+        res = HeadOnCollisionsComparisonWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_accident_type:
+        res = AccidentCountByAccidentTypeWidget(request_params)
+    elif widget_id == WidgetId.accidents_heat_map:
+        res = AccidentsHeatMapWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_accident_year:
+        res = AccidentCountByAccidentYearWidget(request_params)
+    elif widget_id == WidgetId.injured_count_by_accident_year:
+        res = InjuredCountByAccidentYearWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_day_night:
+        res = AccidentCountByDayNightWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_hour:
+        res = AccidentCountByHourWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_road_light:
+        res = AccidentCountByRoadLightWidget(request_params)
+    elif widget_id == WidgetId.top_road_segments_accidents_per_km:
+        res = TopRoadSegmentsAccidentsPerKmWidget(request_params)
+    elif widget_id == WidgetId.injured_count_per_age_group:
+        res = InjuredCountPerAgeGroupWidget(request_params)
+    elif widget_id == WidgetId.vision_zero:
+        res = VisionZeroWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_driver_type:
+        res = AccidentCountByDriverTypeWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_car_type:
+        res = AccidentCountByCarTypeWidget(request_params)
+    elif widget_id == WidgetId.injured_accidents_with_pedestrians:
+        res = InjuredAccidentsWithPedestriansWidget(request_params)
+    elif widget_id == WidgetId.accident_severity_by_cross_location:
+        res = AccidentSeverityByCrossLocationWidget(request_params)
+    elif widget_id == WidgetId.motorcycle_accidents_vs_all_accidents:
+        res = MotorcycleAccidentsVsAllAccidentsWidget(request_params)
+    elif widget_id == WidgetId.accident_count_pedestrians_per_vehicle_street_vs_all:
+        res = AccidentCountPedestriansPerVehicleStreetVsAllWidget(request_params)
+    elif widget_id == WidgetId.top_road_segments_accidents:
+        res = TopRoadSegmentsAccidentsWidget(request_params)
+    elif widget_id == WidgetId.pedestrian_injured_in_junctions:
+        res = PedestrianInjuredInJunctionsWidget(request_params)
+    else:
+        logging.error(f"create_widget: failed to create:{widget_id}")
+        res = None
+    return res
```

Comment:
we can also do a dict mapping between id to function with a dict and use that, just a suggestion.. totally not mendatory.

Reply:
I think we should do this, or something else that is requires less maintainance.

Reply:
At the very least, consider moving this function up, near the enum `WidgetId`.

Reply:
I agree

Reply:
Done. It needs to be after all the widget classes are defined though. (I will push a group of changes together later)

Reply:
>It needs to be after all the widget classes are defined

Not necessarily.